pkg/apis/aws/types_worker.go

```diff
@@ -117,3 +119,24 @@ const (
 	// VolumeTypeGP3 is a constant for the gp3 volume type.
 	VolumeTypeGP3 VolumeType = "gp3"
 )
+
+// HTTPTokensValue is a constant for HTTPTokens values.
+type HTTPTokensValue string
+
+const (
+	// HTTPTokensRequired is a constant for requiring the use of tokens to access IMDS. Effectively disables access via
+	// the IMDSv1 endpoints.
+	HTTPTokensRequired HTTPTokensValue = "required"
+	// HTTPTokensOptional that makes the use of tokens for IMDS optional. Effectively allows access via both IMDSv1 and
+	// IMDSv2 endpoints.
+	HTTPTokensOptional HTTPTokensValue = "optional"
+)
+
+// InstanceMetadataOptions contains configuration for controlling access to the metadata API.
+type InstanceMetadataOptions struct {
```

Comment:
To completely disable IMDS access AWS provides `HttpEndpoint=disabled`. Would it make sense to have that explicitly instead of depending upon the combination of `HttpTokens = required` and `HttpPutResponseHopLimit = 1 or not set`? From the consumer's perspective it becomes simpler to disable via one field.

Reply:
I wrote about this in the special notes:

> Completely disabling access to the metadata service is currently not supported because nodes cannot be created without it. I am not aware of where the problem lies.

If I pass `HttpEndpoint=disabled` to MCM, then the machine cannot come up. So you need to have access to some form of IMDS on the host at the moment. The combination of `httpTokens` and `HttpPutResponseHopLimit` allows you to have access from the host (and therefore the machine can come up), but not from inside containers.

Reply:
> If I pass HttpEndpoint=disabled to MCM, then the machine cannot come up. So you need to have access to some form of IMDS on the host at the moment. 

@himanshu-kun is that the case? I do not see any code in the provider during creation of the VM which mandates this. We can try and check if this is true.

Reply:
@unmarshall it would be nice if you independently validate that. My impression was that it should work without IMDS but somehow in practice it didn't. I would be happy to add the `httpEndpoint` once this point is clarified.